### PR TITLE
File conversion

### PR DESCRIPTION
**Pull request by taylenb**

### CODE REVIEW

File: file_processing/msg_processor.py [hunk 1]
```diff
@@ -21,4 +22,20 @@ def save(self, output_path: str = None) -> None:
...
+    def save_as_pdf(self, output_path: str) -> None:
+        pdf = FPDF()
+        pdf.add_page()
+        pdf.set_font("helvetica", size = 12)
+        msg_data = []
+        msg_data.append([self.metadata['sender'], self.metadata['date'], self.metadata['subject'], self.metadata['text']])
+
+        for x in msg_data:
+            x = str(x)
+            pdf.multi_cell(200,10, txt = x, align = 'C')
```
Comment (pgaviganHC) [inline file_processing/msg_processor.py:40]: What is the meaning of 200 and 10? Be careful with magic numbers.

File: file_processing/pptx_processor.py [hunk 2]
```diff
@@ -27,6 +28,24 @@ def save(self, output_path: str = None) -> None:
         save_path = output_path or self.file_path
         ppt.save(save_path)
 
+
+    def save_as_pdf(self, output_path: str = None) -> None:
+        import comtypes.client
+        import os
+        import sys
+        input = sys.argv[1]
+        output = sys.argv[2]
```
Comment (pgaviganHC) [inline file_processing/pptx_processor.py:37]: What is sys.argv[1] and [2]? The magic numbes make it hard to know.

File: file_processing/pptx_processor.py [hunk 3]
```diff
@@ -27,6 +28,24 @@ def save(self, output_path: str = None) -> None:
...
+
+        input = os.path.abspath(self.file_path)
+        output = os.path.abspath(output_path)
+        powerpoint = comtypes.client.CreateObject("Powerpoint.Application")
+
+        slides = powerpoint.Presentations.Open(input)
+
+        slides.SaveAs(output, 32)
```
Comment (pgaviganHC) [inline file_processing/pptx_processor.py:45]: What does 32 represent here?

File: file_processing/pptx_processor.py [hunk 4]
```diff
@@ -44,4 +63,4 @@ def extract_text_from_pptx(ppt: Presentation) -> str:
             return '\n'.join(full_text)
         except Exception as e:
             print(f"Error encountered while opening or processing {file_path}: {e}")
```
Comment (pgaviganHC) [inline file_processing/pptx_processor.py:65]: Can prints be replaced with logging.error() instead?

**Comment by pgaviganHC** [comment]
I like that the test cases clean up after themselves and don't leave artifacts. Another way to do this is to use test fixtures (https://stackoverflow.com/questions/22627659/run-code-before-and-after-each-test-in-py-test).

**Comment by pgaviganHC** [comment]
@BrennanHCSC, @benjaminLuoHC : Is this PR still active or should it be closed?

**Comment by benjaminLuoHC** [comment]
It seems this PR of old aims to convert between file types. It's an interesting feature, but I feel there's too many combinations of file types to make it flexible enough (ex jpg/jpeg/heic/heif/png file types would have to be convertible to each other, and same for document-based files)

It would also introduce several more external libraries to abstract the conversion process which would make the file processing library less light-weight for users only interested in metadata extraction (though, the same can be argued for the file similarity feature)

**Comment by pgaviganHC** [comment]
This PR is stale. Recommend adding file type conversion as an issue for future feature development if needed.

**Pull request closed without merging**